packages/ember-application/tests/system/application_test.js

```diff
@@ -24,389 +24,404 @@ import {
-  verifyInjection(application, 'renderer', 'env', 'service:-glimmer-environment');
-  verifyRegistration(application, 'view:-outlet');
-  verifyRegistration(application, 'renderer:-dom');
-  verifyRegistration(application, 'renderer:-inert');
-  verifyRegistration(application, P`template:components/-default`);
-  verifyRegistration(application, 'template:-outlet');
-  verifyInjection(application, 'view:-outlet', 'template', 'template:-outlet');
-  verifyInjection(application, 'template', 'env', 'service:-glimmer-environment');
-  deepEqual(application.registeredOptionsForType('helper'), { instantiate: false }, `optionsForType 'helper'`);
-});
+moduleFor('Ember.Application, default resolver with autoboot', class extends AutobootApplicationTestCase {
 
-const originalLogVersion = ENV.LOG_VERSION;
+  constructor() {
+    super();
+    this.originalLookup = context.lookup;
+  }
 
-QUnit.module('Ember.Application initialization', {
   teardown() {
-    if (app) {
-      run(app, 'destroy');
-    }
+    context.lookup = this.originalLookup;
+    super.teardown();
     setTemplates({});
-    ENV.LOG_VERSION = originalLogVersion;
   }
-});
-
-QUnit.test('initialized application goes to initial route', function() {
-  run(() => {
-    app = Application.create({
-      rootElement: '#qunit-fixture'
-    });
-
-    app.Router.reopen({
-      location: 'none'
-    });
-
-    app.register('template:application',
-      compile('{{outlet}}')
-    );
-
-    setTemplate('index', compile(
-      '<h1>Hi from index</h1>'
-    ));
-  });
 
-  equal(jQuery('#qunit-fixture h1').text(), 'Hi from index');
-});
+  createApplication(options) {
+    let myOptions = assign({
+      Resolver: DefaultResolver
+    }, options);
+    return super.createApplication(myOptions);
+  }
 
-QUnit.test('ready hook is called before routing begins', function() {
-  expect(2);
+  [`@test acts like a namespace`](assert) {
+    let lookup = context.lookup = {};
 
-  run(() => {
-    function registerRoute(application, name, callback) {
-      let route = EmberRoute.extend({
-        activate: callback
-      });
+    run(() => {
+      lookup.TestApp = this.createApplication();
+    });
 
-      application.register('route:' + name, route);
-    }
+    setNamespaceSearchDisabled(false);
+    let Foo = this.application.Foo = EmberObject.extend();
+    assert.equal(Foo.toString(), 'TestApp.Foo', 'Classes pick up their parent namespace');
+  }
 
-    let MyApplication = Application.extend({
-      ready() {
-        registerRoute(this, 'index', () => {
-          ok(true, 'last-minute route is activated');
-        });
-      }
+  [`@test can specify custom router`](assert) {
+    let MyRouter = Router.extend();
+    run(() => {
+      let app = this.createApplication();
+      app.Router = MyRouter;
     });
+    assert.ok(
+      this.application.__container__.lookup('router:main') instanceof MyRouter,
```

Comment:
Would be nice to remove this usage of `__container__` (you can use `__deprecatedInstance__` instead for now), as it should be issuing a deprecation...